addons/point_of_sale/models/pos_order.py

```diff
@@ -960,24 +960,25 @@ def set_pack_operation_lot(self, picking=None):
                             if stock_production_lot.product_id.tracking == 'lot':
                                 qty = abs(pos_pack_lot.pos_order_line_id.qty)
                             qty_done += qty
-                            pack_lots.append({'lot_id': stock_production_lot.id, 'qty': qty})
+                            pack_lots.append({'lot': stock_production_lot, 'qty': qty})
                         else:
                             has_wrong_lots = True
                 elif move.product_id.tracking == 'none' or not lots_necessary:
                     qty_done = move.product_uom_qty
                 else:
                     has_wrong_lots = True
                 for pack_lot in pack_lots:
-                    lot_id, qty = pack_lot['lot_id'], pack_lot['qty']
+                    lot, qty = pack_lot['lot'], pack_lot['qty']
+                    quant = lot.quant_ids.filtered(lambda q: q.location_id in move.location_id.child_ids)[-1]
```

Comment:
- [-1] will fail in case there isn't any quant. use [-1:] instead
- the filtered doesn't seems correct, it will only check in the sub-location and not in sub-sub-...-location. Also maybe avoid to take from a negative quant. something like that: 
```python
filtered(lambda q: q.quantity > 0.0 or q.location_id.parent_path.startwith(move.location_id.parent_path))
```